src/map_system.c

```diff
@@ -0,0 +1,1076 @@
+                                 aborting. */
+
+/* map decorator stack */
+//static MapDecorator* decorator_stack = NULL; /**< Contains all the map decorators. */
+//static int decorator_nstack       = 0; /**< Number of map decorators in the stack. */
+
+static StarSystem *cur_sys_sel = NULL; /**< Currently selected system */
+static int cur_planet_sel = 0; /**< Current planet selected by user (0 = star). */
+static Planet *cur_planetObj_sel = NULL;
+static int pitch = 0; /**< pitch of planet images. */
+static int nameWidth = 0; /**< text width of planet name */
+static int nshow = 0; /**< number of planets shown. */
+static char infobuf[PATH_MAX];
+static unsigned int starCnt = 1;
+glTexture **bgImages; /**< nebula and star textures */
+unsigned int nBgImgs; /** number of images */
+
+/* VBO. */
+//static gl_vbo *map_vbo = NULL; /**< Map VBO. */
+//static gl_vbo *marker_vbo = NULL;
+
+#define MAP_SYSTEM_WDWNAME "System map"
+#define MAPSYS_OUTFITS "mapSysOutfits"
+#define MAPSYS_SHIPS "mapSysShips"
+#define MAPSYS_TRADE "mapSysTrade"
+
+/*
+ * extern
+ */
+/* space.c */
+extern StarSystem *systems_stack;
+extern int systems_nstack;
+extern int faction_nstack;
+
+/*land.c*/
+extern int landed;
+extern Planet* land_planet;
+
+/*
+ * prototypes
+ */
+void map_system_close( unsigned int wid, char *str );
+
+int map_system_isOpen (void);
+
+
+
+
+/* Update. */
+//static void map_system_update( unsigned int wid );
+void map_system_updateSelected(unsigned int wid);
+
+/* Render. */
+static void map_system_render( double bx, double by, double w, double h, void *data );
+/* Mouse. */
+static int map_system_mouse( unsigned int wid, SDL_Event* event, double mx, double my,
+      double w, double h, void *data );
+/* Misc. */
+static int map_system_keyHandler( unsigned int wid, SDL_Keycode key, SDL_Keymod mod );
+//static void map_system_selectCur (void);
+void map_system_show( int wid, int x, int y, int w, int h);
+
+static void map_system_genOutfitsList( unsigned int wid, float goodsSpace, float outfitSpace, float shipSpace );
+static void map_system_genShipsList( unsigned int wid, float goodsSpace, float outfitSpace, float shipSpace );
+static void map_system_genTradeList( unsigned int wid, float goodsSpace, float outfitSpace, float shipSpace );
+
+
+static void map_system_array_update( unsigned int wid, char* str );
+static void map_system_array_rmouse( unsigned int wid, char* widget_name );
+
+void map_system_buyCommodPrice( unsigned int wid, char *str );
+
+/**
+ * @brief Initializes the map subsystem.
+ *
+ *    @return 0 on success.
+ */
+int map_system_init (void)
+{
+   return 0;
+}
+
+/**
+ * @brief Placemarker for when required.
+ *
+ *    @return 0 on success.
+ */
+int map_system_load (void)
+{
+  return 0;
+}
+
+/**
+ * @brief Destroys the map subsystem.
+ */
+void map_system_exit (void)
+{
+   for( unsigned int i=0; i<nBgImgs; i++ ){
+      gl_freeTexture ( bgImages[i] );
```

Comment:
OK, so, a couple style comments (these won't prevent a merge but just want to let you know):

* `for` statements (and other things that aren't functions) should have a space between them and the parentheses following them.
* Function calls (such as `glx_freeTexture` here) should have the opening parenthesis immediately after, with no space.
* This one's minor, but there should be a space between a closing parenthesis and an opening angle bracket.

So this particular section of code should instead be:

```
   for ( unsigned int i=0; i<nBgImgs; i++ ) {
      gl_freeTexture( bgImages[i] );
```

Reply:
Yeah, thanks - that's mostly the tidying up I was talking about - old habbits die hard (what I started programming, I was saving to floppy disks, and every byte counted, as I had to buy them with my pocket money!